src/i2s.rs

```diff
@@ -26,6 +26,12 @@ pub trait Instance:
 {
 }
 
+/// Trait for SPI peripheral that have an extension for full duplex i2s capability.
+pub trait Extension: gpio::alt::I2sExtPin {
+    /// The I2SEXT peripheral that extend the SPI peripheral
+    type I2sExtPeripheral;
+}
+
```

Comment:
Possibly it is better to depend `DualI2sPeripheral` on `I2sPeripheral` and same:

```suggestion
pub trait DualInstance: Instance + gpio::alt::I2sExtPin {
    type I2sExtPeripheral;
}

```